tla/consensus/ccfraft.tla

```diff
@@ -848,7 +931,15 @@ ConflictAppendEntriesRequest(i, index, m) ==
        IN /\ log' = [log EXCEPT ![i] = new_log]
           \* Potentially also shorten the configurations if the removed txns contained reconfigurations
           /\ configurations' = [configurations EXCEPT ![i] = ConfigurationsToIndex(i,Len(new_log))]
-    /\ UNCHANGED <<removedFromConfiguration, serverVars, commitIndex, messages, membershipState>>
+          \* Potentially rollback retirement phase
+          /\ IF /\ membershipState[i] \in {RetirementOrdered, RetirementSigned}
+                /\ RetirementIndex(i) > Len(log'[i])
```

Comment:
Should `RetirementIndex` also be primed to be calculated on the new log?

Reply:
Using retirementIndex of log is ok in this very particular case but I agree that it's unclear so I've refactored in https://github.com/microsoft/CCF/pull/5919/commits/c3f7fb3d883e35d97566d57fee5bf6ed7a470d43.